node/src/components/consensus/highway_core/state/params.rs

```diff
@@ -121,6 +123,11 @@ impl Params {
     pub(crate) fn endorsement_evidence_limit(&self) -> u64 {
         self.endorsement_evidence_limit
     }
+
+    /// Returns the minimum lenght of the era.
+    pub(crate) fn min_era_length(&self) -> TimeDiff {
+        TimeDiff::from(1 << self.min_round_exp) * self.end_height
```

Comment:
Is this necessary though? Can an era be shorter than the minimum number of rounds with the lowest round exponent possible?

Reply:
No, but it can be impossible that it's shorter.
If we set a minimum height to 10 and minimum duration to 0, it'll return the round-based expression. If we set minimum height to 0 and minimum duration to 10 minutes, it _should_ return 10 minutes.

Reply:
Ah, I forgot that it's _either or_. The `end_height` is misnomer as the value passed in is relative, not absolute (i.e. it's height difference and not final block height).

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/913/commits/faaa85ffb52bc7423f03dd9711c3211f894f9432.

Reply:
Yes, inside `highway_core` everything is relative, and the concept of "era" doesn't really exist. (So maybe this method should be named `min_duration`?)

Reply:
Again, I forgot that every era starts numbering from 0.